src/model/analysis.ts

```diff
@@ -23,6 +23,11 @@ import {ResolvedUrl} from './url';
 import {Warning} from './warning';
 
 
+export type Result<G, B> = {
```

Comment:
If this isn't used anywhere else, it might be nice to make it an `AnalysisResult` with `.document` and `.error`.

Reply:
I'd like to swap this out for a Result class later, to add useful methods.